src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkModal.tsx

```diff
@@ -1,37 +1,127 @@
-import React, { useState } from "react"
-import { Spacer, Join, ModalDialog } from "@artsy/palette"
-import {
-  ListItemEntity,
-  SelectListItem,
-} from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListItem"
-import { SelectListsForArtworkHeader } from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkHeader"
+import React, { FC, useState } from "react"
+import { Spacer, Join, ModalDialog, Box } from "@artsy/palette"
+import { SelectListItemFragmentContainer } from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListItem"
+import { SelectListsForArtworkHeaderFragmentContainer } from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkHeader"
 import { SelectListsForArtworkFooter } from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkFooter"
-import { times } from "lodash"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { createFragmentContainer, graphql } from "react-relay"
+import { SelectListsForArtworkModal_me$data } from "__generated__/SelectListsForArtworkModal_me.graphql"
+import { SelectListsForArtworkModal_artwork$data } from "__generated__/SelectListsForArtworkModal_artwork.graphql"
+import { SelectListsForArtworkModalQuery } from "__generated__/SelectListsForArtworkModalQuery.graphql"
+import { extractNodes } from "Utils/extractNodes"
+import {
+  SelectListsForArtworkHeaderPlaceholder,
+  SelectListsPlaceholder,
+} from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkPlaceholders"
+import { getSelectedCollectionIds } from "Apps/CollectorProfile/Routes/Saves2/Utils/getSelectedCollectionIds"
+
+interface SelectListsForArtworkModalQueryRenderProps {
+  artworkID: string
+  onClose: () => void
+}
 
 export interface SelectListsForArtworkModalProps {
+  me: SelectListsForArtworkModal_me$data | null
+  artwork: SelectListsForArtworkModal_artwork$data | null
   onClose: () => void
 }
 
 export const SelectListsForArtworkModal: React.FC<SelectListsForArtworkModalProps> = ({
```

Comment:
There are some questions and difficulties with displaying the modal for mobile devices. I want to clarify and discuss all the points with Ryan. If changes are required, I will add them in a separate PR